unified-codes/apps/data-service/src/app/v1/types/DgraphDataSource.ts

```diff
@@ -164,7 +165,16 @@ export class DgraphDataSource extends RESTDataSource {
     const data = await this.postQuery(DgraphDataSource.getEntityQuery(code));
 
     const { query } = data ?? {};
-    const [entity] = query ?? [];
+    const [entity]: [IEntity] = query ?? [];
+    entity.type = DgraphDataSource.getEntityType(entity);
+
+    const mapChild = (child: IEntity) => {
+      child.type = DgraphDataSource.getEntityType(child);
+      child.children && child.children.forEach(mapChild);
+    };
+
+    entity.children.forEach(mapChild);
```

Comment:
Think could simplify this a little bit:

```
const mapEntity = (entity: IEntity) => {
  const type = DgraphDataSource.getEntityType(entity);
  const children = entity.children.map(child => mapEntity(child));
  return { ...entity, type, children };
}

const updatedEntity = mapEntity(entity);
```

Prob can come up with better variable ideas!

Reply:
Ohh. Thanks yep that's much nicer. Yoink. 07ff06374